lib/transaction/transaction.js

```
@@ -334,7 +334,7 @@ Transaction.prototype.fromBufferReader = function(reader) {
   }
   this.nLockTime = reader.readUInt32LE();
 
-  if (this.isSpecialTransaction()) {
+  if (this.isSpecialTransaction() && !reader.finished()) {
```

What happens now if the payload was omitted? Expected behavior would be to error out in some way and it should not silently succeed

---

@codablock 
It isn't silently succeeding. Parsing and validation are just happens in different methods. Actual validation happens here: https://github.com/dashevo/dashcore-lib/blob/feature/DIP2/lib/transaction/transaction.js#L211
It throws in three cases:
1. A type is set, but a payload is not set;
2. The type and the payload are set, but payload size in a serialized transaction doesn't match an actual payload size;
3. The payload is set, but the type is not set